src/promptflow/promptflow/_sdk/operations/_flow_conext_resolver.py

```diff
@@ -0,0 +1,121 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+from functools import lru_cache
+from os import PathLike
+from pathlib import Path
+from typing import Dict
+
+from promptflow._sdk._constants import NODES
+from promptflow._sdk._utils import get_local_connections_from_executable, parse_variant
+from promptflow._sdk.entities import FlowContext
+from promptflow._sdk.operations._run_submitter import (
+    _load_flow_dag,
+    overwrite_connections,
+    overwrite_flow,
+    overwrite_variant,
+)
+from promptflow.contracts.flow import Flow as ExecutableFlow
+from promptflow.contracts.flow import Node
+from promptflow.exceptions import UserErrorException
+from promptflow.executor import FlowExecutor
+
+# Resolve flow context to executor
+# Resolve flow according to flow context
+#   Resolve connection, variant, overwrite, store in-memory
+# create executor based on resolved flow
+# cache executor if flow context not changed (define hash function for flow context).
+
+
+class FlowContextResolver:
+    """Flow context resolver."""
+
+    def __init__(self, flow_path: PathLike):
+        from promptflow import PFClient
+
+        self.flow_path, self.flow_dag = _load_flow_dag(flow_path=Path(flow_path))
+        self.working_dir = Path(self.flow_path).parent.resolve()
+        self.node_name_2_node: Dict[str, Node] = {node["name"]: node for node in self.flow_dag[NODES]}
+        self.client = PFClient()
+
+    @classmethod
+    @lru_cache
```

Comment:
This is a bit dangerous, maybe cache on the load_flow() obj, 
maybe what if user change the flow.dag.yaml run a notebook cell, but did not see an update there?

Reply:
maybe we can change the interface to 

```python
def create(cls, flow: Flow) -> FlowExecutor:
```

Reply:
As offline synced, changed to above interface.